reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -598,14 +674,17 @@ def insert_slice(slice_real, xyz, n_pix):
             otherwise 0.
             Shape (n_pix, n_pix, n_pix)
         """
-        inserted_slice_tensor = torch.sparse_coo_tensor(
-            xyz + n_pix // 2, slice_real.reshape((n_pix**2,)), (n_pix, n_pix, n_pix)
-        )
-        count_tensor = torch.sparse_coo_tensor(
-            xyz + n_pix // 2, np.ones((n_pix**2,)), (n_pix, n_pix, n_pix)
-        )
-        inserted_slice_3d = inserted_slice_tensor.to_dense().numpy()
-        count_3d = count_tensor.to_dense().numpy()
+        n_pix = slice_real.shape[0]
+        slice_values = np.tile(slice_real.reshape((n_pix**2,)), (3,))
```

Comment:
Doesn't it work if you just don't tile, and input a regular on offset rotated plane? If it doesn't there are other functions that can do the interpolation, besides `griddata`. It's not needed in principle. See for example https://github.com/geoffwoollard/learn_cryoem_math/blob/master/nb/fourier_slice_2D_3D_with_trilinear.ipynb